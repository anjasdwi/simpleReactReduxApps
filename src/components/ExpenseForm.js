import React from 'react';
// import 'react-dates/initialize'
import moment from 'moment'
import { SingleDatePicker } from 'react-dates';
// import 'react-dates/lib/css/_datepicker.css'


// const now = moment();
// console.log(now.format('MMM Do, YYYY'));

class ExpenseFrom extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            description: props.expense ? props.expense.description : '',
            note : props.expense ? props.expense.note : '',
            amount: props.expense ? (props.expense.amount / 100).toString() : '',
            createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
            calFocused : false,
            error : ''
        };
        // console.log(props.expense ? (props.expense.amount / 100).toString() : '');
    }
    
    onDescription = (e) => {
        const description = e.target.value;
        this.setState(()=> ({ description }))
    }

    onNote = (e) => {
        const note = e.target.value;
        this.setState(()=> ({ note }))
    }

    onAmount = (e) => {
        const amount = e.target.value;
        if (amount.match(/^\d*(\.\d{0,2})?$/)) {
            this.setState(()=> ({ amount }));
        }
    }

    onDate = (createdAt) => {
        if (createdAt) {
            this.setState(() =>({ createdAt }) )
        }
    }

    onFocusChange = ({ focused }) => {
        this.setState(()=> ({calFocused: focused}));
    }

    onSubmit= (e) => {
        e.preventDefault()
        if (!this.state.description || !this.state.amount) {
            this.setState(()=>({ error : 'Mohon lengkapi description and amount'}));
        } else {
            this.setState(()=> ({ error: ''}));
            this.props.onSubmit({
                description: this.state.description,
                amount: parseFloat(this.state.amount, 10) * 100,
                createdAt: this.state.createdAt.valueOf(),
                note: this.state.note
            })
            // console.log(this.state);
        }
    }
    render () {
        return (
            <div>
                {this.state.error && <p>{this.state.error}</p>}
                <form onSubmit={this.onSubmit}>
                    <input type="text"
                    placeholder="Description"
                    autoFocus
                    value={this.state.description}
                    onChange= {this.onDescription}
                    />
                    <input
                        type="text"
                        placeholder="Amount"
                        onChange={this.onAmount}
                        value={this.state.amount}
                    />
                    <SingleDatePicker
                    date={this.state.createdAt} 
                    onDateChange={this.onDate} 
                    focused={this.state.calFocused}
                    onFocusChange={this.onFocusChange}
                    numberOfMonths={1}
                    isOutsideRange={()=>false}
                    />
                    <textarea
                    placeholder="Add your Note"
                    value={this.state.note}
                    onChange= {this.onNote}
                    >
                    </textarea>
                    <button>Add Expense</button>
                </form>
            </div>
        )
    }
}

export default ExpenseFrom