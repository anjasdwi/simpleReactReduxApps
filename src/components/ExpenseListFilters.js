import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';


export class ExpenseListFilters extends React.Component {
    
    state = {
        calendarFocused: null
    }
    onDatesChange = ({ startDate, endDate}) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    }
    onFocusChange = (calendarFocused) => {
        this.setState(()=>({ calendarFocused }));
        
    }
    render() {
        console.log(this.props.filters.startDate)
        return  (
            <div>
                <input type="text" value={this.props.filters.text} onChange={(e)=> {
                    this.props.setTextFilter(e.target.value)
                }}/>
                <select value={this.props.filters.sortBy}
                onChange={(e)=> {
                    if (e.target.value === 'date') {
                        this.props.sortByDate();
                    }else if (e.target.value === 'amount') {
                        this.props.sortByAmount();
                    }
                }}>
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>
                <DateRangePicker 
                startDate={this.props.filters.startDate}
                startDateId={"start"}
                endDateId={"end"}
                endDate={this.props.filters.endDate}
                onDatesChange={this.onDatesChange}
                focusedInput={this.state.calendarFocused}
                showClearDates={true}
                onFocusChange={this.onFocusChange}
                numberOfMonths={1}
                isOutsideRange={()=> false}
                />
            </div>
        )
    }
}

const mapStateToProps = (state) => ({
    filters: state.filters
  });
  
const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount()),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate))
});

export default  connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);