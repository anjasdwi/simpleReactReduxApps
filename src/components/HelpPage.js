import React from 'react';

const HelpPage =  () => (
    <div>
        Dashboard Help
    </div>
)

export default HelpPage