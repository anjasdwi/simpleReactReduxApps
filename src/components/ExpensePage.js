import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { addExpense } from '../actions/expenses'

export class ExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.onSubmit(expense);
        this.props.history.push('/');
    };
    render() {
        return (
            <div>
                <h1>Add Expense</h1>
                <div></div>
                <ExpenseForm onSubmit={this.onSubmit}/>
            </div>
        )
    }
}
            

const mapDispatchToProps = (dispatch) => ({
        onSubmit: (expense) => dispatch(addExpense(expense))
})

export default connect(undefined, mapDispatchToProps)(ExpensePage);

  