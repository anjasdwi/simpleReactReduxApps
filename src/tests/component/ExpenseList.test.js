import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { ExpenseList } from '../../components/ExpenseList'
import expenses from '../fixtures/expenses';

test('Should render Expenselist with', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
})

test('Should render Expenselist with', () => {
    const wrapper = shallow(<ExpenseList expenses={[]} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
})
