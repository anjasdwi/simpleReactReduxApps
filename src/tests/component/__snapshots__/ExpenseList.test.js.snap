// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Expenselist with 1`] = `
<div>
  <ExpenseListItem
    amount={195}
    createdAt={0}
    description="Gum"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={10950}
    createdAt={-345600000}
    description="Rent"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={4500}
    createdAt={345600000}
    description="Credit Card"
    id="3"
    key="3"
    note=""
  />
</div>
`;

exports[`Should render Expenselist with 2`] = `
<div>
  <p>
    No Expenses
  </p>
</div>
`;
